solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -1,26 +1,78 @@
+use std::str::FromStr;
+
+use anchor_lang::prelude::{CpiContext, Pubkey, AccountInfo};
 use anchor_lang::solana_program::msg;
+use anchor_lang::solana_program::pubkey::ParsePubkeyError;
+use anchor_spl::token::{spl_token, Burn, MintTo, Transfer};
 use ibc::applications::transfer::context::{
     TokenTransferExecutionContext, TokenTransferValidationContext,
 };
 use ibc::applications::transfer::error::TokenTransferError;
-use ibc::applications::transfer::PrefixedCoin;
+use ibc::applications::transfer::{PrefixedCoin, Amount};
 use ibc::core::ics24_host::identifier::{ChannelId, PortId};
-use ibc::Signer;
+use uint::FromDecStrErr;
+use primitive_types::U256;
 
 // use crate::module_holder::IbcStorage<'_,'_>;
-use crate::IbcStorage;
+use crate::{IbcStorage, MINT_ESCROW_SEED};
+
+pub struct InnerPubkey(pub Pubkey);
+
+impl TryFrom<ibc::Signer> for InnerPubkey {
+
+   type Error = ParsePubkeyError; 
 
-impl TokenTransferExecutionContext for IbcStorage<'_, '_> {
+   fn try_from(value: ibc::Signer) -> Result<Self, Self::Error> {
+        Ok(InnerPubkey(Pubkey::from_str(&value.to_string())?))
+   } 
+}
+
```

Comment:
Some further suggestions regarding the type.  You could also implement
PartialEq:

```
impl PartialEq<Pubkey> for InnerPubkey {
    #[inline]
    fn eq(&self, rhs: &Pubkey) -> bool { &self.0 == rhs }
}

impl PartialEq<InnerPubkey> for Pubkey {
    #[inline]
    fn eq(&self, rhs: &InnerPubkey) -> bool { self == &rhs.0 }
}
```

This way you won’t need to access the wrapped value when comparing.

You can also further add some more derives:

```
#[derive(Clone, PartialEq, Eq, derive_more::From, derive_more::Into)]
#[into(owned, ref, ref_mut)]
pub struct AccountId(Pubkey);
```

Perhaps I’m just being pedantic, but to me `Pubkey::from(account)`
reads better than `account.0`.

If you really want to go to town with this type, you can also do:

```
impl core::fmt::Debug for AccountId {
    #[inline]
    fn fmt(&self, fmtr: &mut core::fmt::Formatter) -> core::fmt::Result {
        self.0.fmt(fmtr)
    }
}

impl core::fmt::Display for AccountId {
    #[inline]
    fn fmt(&self, fmtr: &mut core::fmt::Formatter) -> core::fmt::Result {
        self.0.fmt(fmtr)
    }
}
```

and you’d probably never need `.0`.

Reply:
But `Pubkey::from(account)` would require me to implement a trait on Pubkey but since it is from external crate, it wouldnt be possible for me to do it. I can convert from `Pubkey` to `AccountId`, not the other way around i suppose.

Reply:
But if you wanna remove the `.0`, what do u think about this?
```suggestion
impl AccountId {
    fn to_pubkey(self) -> Pubkey {
        self.0
    }
}
```

Reply:
> But Pubkey::from(account) would require me to implement a trait on Pubkey but since it is from external crate,

That’s fine. You can implement `From<InternalType> for ExternalType` traits.  `derive_more::Into` will do that for you.

Reply:
alright